botocore/credentials.py

```diff
@@ -477,6 +454,30 @@ def get_frozen_credentials(self):
         return self._frozen_credentials
 
 
+class DeferredRefreshableCredentials(RefreshableCredentials):
+    """Refreshable credentials that don't require initial credentials.
+
+    refresh_using will be called upon first access.
+    """
+    def __init__(self, refresh_using, method, time_fetcher=_local_now):
+        self._refresh_using = refresh_using
+        self._access_key = None
+        self._secret_key = None
+        self._token = None
+        self._expiry_time = None
+        self._time_fetcher = time_fetcher
+        self._refresh_lock = threading.Lock()
+        self.method = method
+        self._frozen_credentials = None
```

Comment:
Just thinking out loud here. I understand why you didn't just use super here because you do not want `_normalize()` to be called. I'm a little concerned that these internal variables may grow out of sync with the `RefreshableCredentials`, but given that we hardly touch this code and we now have a lot of test, I'm not too concerned about it.